playback/service/src/main/java/de/danoeh/antennapod/playback/service/internal/EpisodeSleepTimer.java

```diff
@@ -0,0 +1,74 @@
+package de.danoeh.antennapod.playback.service.internal;
+
+import android.content.Context;
+
+import org.greenrobot.eventbus.EventBus;
+
+import java.util.concurrent.TimeUnit;
+
+import de.danoeh.antennapod.event.playback.PlaybackPositionEvent;
+import de.danoeh.antennapod.event.playback.SleepTimerUpdatedEvent;
+
+public class EpisodeSleepTimer extends ClockSleepTimer {
+    private static final String TAG = "EpisodeSleepTimer";
+
+    public EpisodeSleepTimer(final Context context) {
+        super(context);
+    }
+
+    @Override
+    public boolean isEndingThisEpisode(long episodeRemainingMillis) {
+        return getTimeLeft().getDisplayValue() == 1;
+    }
+
+    @Override
+    public TimerValue getTimeLeft() {
+        TimerValue x = super.getTimeLeft();
+        return new TimerValue(x.getDisplayValue(), TimeUnit.DAYS.toMillis(x.getDisplayValue()));
+    }
+
+    @Override
+    public void playbackPositionUpdate(PlaybackPositionEvent playbackPositionEvent) {
+        long currentEpisodeTimeLeft = playbackPositionEvent.getDuration() - playbackPositionEvent.getPosition();
+
+        final TimerValue current = getTimeLeft();
+
+        if (isEndingThisEpisode(playbackPositionEvent.getPosition())) {
```

Comment:
The function accepts remaining millis, not the position. However, doing this with the position/duration is difficult because of "skip silence", playback speed, and "skip ending". I would just ignore the volume reduction for episode sleep timers